### PR TITLE
netspeed: Add netspeed-preferences.ui

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: netspeed/src/netspeed.c [hunk 1]
```diff
@@ -971,62 +952,22 @@ settings_cb(GtkAction *action, gpointer data)
...
-	gtk_label_set_justify(GTK_LABEL(network_device_label), GTK_JUSTIFY_LEFT);
-	gtk_label_set_xalign (GTK_LABEL (network_device_label), 0.0f);
-	gtk_label_set_yalign (GTK_LABEL (network_device_label), 0.5f);
-	gtk_size_group_add_widget(category_label_size_group, network_device_label);
-	gtk_box_pack_start(GTK_BOX (network_device_hbox), network_device_label, FALSE, FALSE, 0);
-
-	applet->network_device_combo = gtk_combo_box_text_new();
-	gtk_label_set_mnemonic_widget(GTK_LABEL(network_device_label), applet->network_device_combo);
-	gtk_box_pack_start (GTK_BOX (network_device_hbox), applet->network_device_combo, TRUE, TRUE, 0);
+	g_settings_bind (applet->gsettings, "show-sum", gtk_builder_get_object (builder, "show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
```
Comment (tamplan) [inline netspeed/src/netspeed.c:965]: Hi,

Only for me i understand the right way to do, is there correct to use GET_WIDGET("show_sum_checkbutton") here?

Thanks for your reply.
Regards

Comment (rbuj) [reply netspeed/src/netspeed.c:965]: GET_WIDGET & GET_DIALOG are local macros, you can use -E to expand them:
```C
 applet->settings = (((((GtkDialog*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("preferences_dialog")))), ((gtk_dialog_get_type ())))))));
 applet->network_device_combo = (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("network_device_combo")))), ((gtk_widget_get_type ())))))));

 gtk_window_set_screen(((((GtkWindow*) g_type_check_instance_cast ((GTypeInstance*) ((applet->settings)), ((gtk_window_get_type ())))))),
         gtk_widget_get_screen(((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((applet->settings)), ((gtk_widget_get_type ()))))))));

 gtk_dialog_set_default_response(((((GtkDialog*) g_type_check_instance_cast ((GTypeInstance*) ((applet->settings)), ((gtk_dialog_get_type ())))))), GTK_RESPONSE_CLOSE);

 g_settings_bind (applet->gsettings, "show-sum", gtk_builder_get_object (builder, "show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 g_settings_bind (applet->gsettings, "show-bits", gtk_builder_get_object (builder, "show_bits_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 g_settings_bind (applet->gsettings, "short-unit", gtk_builder_get_object (builder, "short_unit_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 g_settings_bind (applet->gsettings, "show-icon", gtk_builder_get_object (builder, "show_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 g_settings_bind (applet->gsettings, "show-quality-icon", gtk_builder_get_object (builder, "show_quality_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 g_settings_bind (applet->gsettings, "change-icon", gtk_builder_get_object (builder, "change_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
```
[gtk_builder_get_object](https://developer.gnome.org/gtk3/stable/GtkBuilder.html#gtk-builder-get-object) returns a `GObject *`, 3rd parameter of [g_settings_bind](https://developer.gnome.org/gio/stable/GSettings.html#g-settings-bind) is a `GObject *`.  GtkWidget is a base class, [GObject](https://developer.gnome.org/gtk3/unstable/GtkWidget.html#GtkWidget.object-hierarchy) is the root class.

Comment (tamplan) [reply netspeed/src/netspeed.c:965]: So, if i have understood (i have read the declaration of the gtk_builder_get_object function in the doc before asking you), you can change:
`g_settings_bind (applet->gsettings, "show-sum", gtk_builder_get_object (builder, "show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);`
To:
`g_settings_bind (applet->gsettings, "show-sum", GET_WIDGET("show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);`

It's only "syntactic sugar" (as saying in french, my primary language) just to win time coding.

Thanks!

Comment (rbuj) [reply netspeed/src/netspeed.c:965]:
```diff
--- a/netspeed/src/netspeed.c
+++ b/netspeed/src/netspeed.c
@@ -962,12 +962,12 @@ settings_cb(GtkAction *action, gpointer data)
 
        gtk_dialog_set_default_response(GTK_DIALOG(applet->settings), GTK_RESPONSE_CLOSE);
 
-       g_settings_bind (applet->gsettings, "show-sum", gtk_builder_get_object (builder, "show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
-       g_settings_bind (applet->gsettings, "show-bits", gtk_builder_get_object (builder, "show_bits_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
-       g_settings_bind (applet->gsettings, "short-unit", gtk_builder_get_object (builder, "short_unit_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
-       g_settings_bind (applet->gsettings, "show-icon", gtk_builder_get_object (builder, "show_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
-       g_settings_bind (applet->gsettings, "show-quality-icon", gtk_builder_get_object (builder, "show_quality_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
-       g_settings_bind (applet->gsettings, "change-icon", gtk_builder_get_object (builder, "change_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "show-sum", GET_WIDGET ("show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "show-bits", GET_WIDGET ("show_bits_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "short-unit", GET_WIDGET ("short_unit_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "show-icon", GET_WIDGET ("show_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "show-quality-icon", GET_WIDGET ("show_quality_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
+       g_settings_bind (applet->gsettings, "change-icon", GET_WIDGET ("change_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
 
        /* Default means device with default route set */
        gtk_combo_box_text_append_text(GTK_COMBO_BOX_TEXT(applet->network_device_combo), _("Default"));
```
Expand macros:
```
gcc -E -DHAVE_CONFIG_H -I. -I../..  -I../.. -I/usr/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -pthread  -I/usr/include/libgtop-2.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -I/usr/include/mate-panel-4.0/libmate-panel-applet -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/at-spi-2.0 -pthread  -DNETSPEED_MENU_UI_DIR=\""/usr/share/mate/ui"\" -Wall -Wmissing-prototypes -Wbad-function-cast -Wcast-align -Wextra -Wno-unused-parameter -Wformat-nonliteral -Wmissing-declarations -Wmissing-field-initializers -Wnested-externs -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-prototypes -Wno-sign-compare   -g -O0 -MT netspeed.o -MD -MP -MF .deps/netspeed.Tpo  netspeed.c
```
diff
```diff
78374,78379c78374,78379
<  g_settings_bind (applet->gsettings, "show-sum", gtk_builder_get_object (builder, "show_sum_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
<  g_settings_bind (applet->gsettings, "show-bits", gtk_builder_get_object (builder, "show_bits_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
<  g_settings_bind (applet->gsettings, "short-unit", gtk_builder_get_object (builder, "short_unit_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
<  g_settings_bind (applet->gsettings, "show-icon", gtk_builder_get_object (builder, "show_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
<  g_settings_bind (applet->gsettings, "show-quality-icon", gtk_builder_get_object (builder, "show_quality_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
<  g_settings_bind (applet->gsettings, "change-icon", gtk_builder_get_object (builder, "change_icon_checkbutton"), "active", G_SETTINGS_BIND_DEFAULT);
---
>  g_settings_bind (applet->gsettings, "show-sum", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("show_sum_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
>  g_settings_bind (applet->gsettings, "show-bits", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("show_bits_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
>  g_settings_bind (applet->gsettings, "short-unit", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("short_unit_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
>  g_settings_bind (applet->gsettings, "show-icon", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("show_icon_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
>  g_settings_bind (applet->gsettings, "show-quality-icon", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("show_quality_icon_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
>  g_settings_bind (applet->gsettings, "change-icon", (((((GtkWidget*) g_type_check_instance_cast ((GTypeInstance*) ((gtk_builder_get_object (builder, ("change_icon_checkbutton")))), ((gtk_widget_get_type ()))))))), "active", G_SETTINGS_BIND_DEFAULT);
```

**Pull request merged**